astropy/io/fits/hdu/hdulist.py

```diff
@@ -797,6 +792,22 @@ def filename(self):
         return None
 
     @classmethod
+    def _ext_match(cls, ext, hdu, idx):
+        if _is_int(ext):
+            return ext == idx
+        # it has to be a tuple now
+        extname, extver = ext
+        if extname is None:
+            return False
+        extname = extname.upper()
```

Comment:
updated to move the the strip and upper into this function instead of being an input requirement of the function + removed the equivalent in one of the callers.
So it makes a bit more work in index_of but it should be not significant, the less error prone api is imo preferable.